README.md

```diff
@@ -56,15 +97,164 @@ This library is a simpler interface to Pelion DM Client, making it trivial to ex
+Note the following:
+ * `mbed-bootloader` is primarily optimized for `GCC_ARM` and therefore you might want to compile it with that toolchain.
+ * Before jumping to the next step, you should compile and flash the bootloader, and then connect over the virtual comport to ensure that the bootloader is running correctly. You can ignore errors related to checksum verification or falure to jump to application - these are expected at this stage.
 
-## Testing
+### 3. Include the bootloader
 
-Simple Pelion DM Client provides Greentea tests to test your porting efforts. Before running these tests, it's recommended that you run the `mbed dm init` command, which will install all needed credentials for both Connect and Update Pelion DM features. You can use the following command:
+1. Copy the compiled bootloader from `mbed-bootloader-extended/BUILDS/<TARGET>/<TOOLCHAIN>-TINY/mbed-bootloader.bin` to `<your_application_name>/bootloader/mbed-bootloader-<TARGET>.bin`.
+
+2. Edit `<your_application_name>/mbed_app.json` and modify the target entry to include:
+```
+            "target.features_add"              : ["BOOTLOADER"],
+            "target.bootloader_img"            : "bootloader/mbed-bootloader-<TARGET>.bin",
+            "target.app_offset"                : "0x10400",
+            "target.header_offset"             : "0x10000",
+            "update-client.application-details": "(MBED_CONF_APP_FLASH_START_ADDRESS + 64*1024)",
+```
+ 
+   Note that:
+   * `update-client.application-details` should be identical in both `bootloader_app.json` and `mbed_app.json`
+   * `target.app_offset` is relative offset to `flash-start-address` you specified in the `mbed_app.json` and `bootloader_app.json`, and is the hex value of the offset specified by `application-start-address` in `bootloader_app.json`, e.g. `(MBED_CONF_APP_FLASH_START_ADDRESS+65*1024)` dec equals `0x10400` hex.
+   * `target.header_offset` is also relative offset to the `flash-start-address` you specified in the `bootloader_app.json`, and is the hex value of the offset specified by `update-client.application-details`, e.g. `(MBED_CONF_APP_FLASH_START_ADDRESS+64*1024)` dec equals `0x10000` hex.
+
+7. Finally, compile and re-run all tests, including the firmware update ones with:
+```
+$ mbed test -t <TOOLCHAIN> -m <BOARD> -n simple*dev*connect -DMBED_TEST_MODE --compile
+
+$ mbed test -t <TOOLCHAIN> -m <BOARD> -n simple*dev*connect --run -v
+```
+
+Refer to the next section about what tests are being executed.
+
+## Validation and Testing
+
+Mbed Device Management provides built-in tests to help you when define your device management configuration. Before running these tests, it's recommended that you run the `mbed dm init` command, which will install all needed credentials for both Connect and Update Pelion DM features. You can use the following command:
 ```
 $ mbed dm init -d "<your company name.com>" --model-name "<product model identifier>" -q --force
 ```
```

Comment:
This line is repeated 3 times in the doc. I think we should remove it and use the one in the 'Testing Setup' section below